src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1630,6 +1619,10 @@ void TorrentHandle::handleMetadataReceivedAlert(libt::metadata_received_alert *p
         m_speedMonitor.reset();
         m_session->handleTorrentPaused(this);
     }
+    
+    // If first/last piece priority was specified when adding this torrent, set it up now that we can:
+    if (m_hasFirstLastPiecePriority)
+        setFirstLastPiecePriority(true);
```

Comment:
But how to resume this mode then. Frankly speaking, I lost both of you in this discussion couple of iterations ago. IMO, the implementation has to be simple:
- add a member flag to `TorrentHandle`: `bool m_needApplyFirstLastPriorityMode;`
- in `TorrentHandle::setFirstLastPiecePriority()` if there are no metadata, set this flag, otherwise proceed as usual and reset the flag;
- in `TorrentHandle::handleMetadataReceivedAlert()` check the flag:  call  `setFirstLastPiecePriority(true);` when it is set.
- save the flag to resume data (probably only if it is set).